client-app/src/desktop/common/Wrapper.js

```diff
@@ -20,10 +54,70 @@ class Wrapper extends Component {
                     marginBottom: 10,
                     omit: !description
                 }),
-                children
+                children,
+                this.renderSourceCodePanel()
             ],
             ...rest
         });
     }
+    
+    renderSourceCodePanel() {
+        const {dockContainerModel} = this;
+        const {links, link} = this.props;
+        
+        if (links || link) {
+            dockContainerModel.addView({
+                id: XH.genId(),
+                icon: Icon.code(),
+                title: code('Source Code'),
+                allowDialog: false,
+                allowClose: false,
+                content: panel({
+                    className: 'toolbox-wrapper-sourcecode',
+                    item: this.generateLinks()
+                })
+            });
+            
+            return dockContainer({
+                model: dockContainerModel
+            });
+        }
+    }
+    
+    generateLinks() {
+        const arrayLinks = this.props.links || castArray(this.props.link);
+        
+        return div(
+            arrayLinks.map(linkObj => this.renderSingleLink(linkObj))
+        );
+        
+    }
+    
+    renderSingleLink(linkObj) {
+        return p(
+            a({
+                href: this.generateUrl(linkObj.url),
+                item: code(linkObj.text),
+                target: '_blank'
+            }),
+            ' | ',
```

Comment:
Minor thing, but I think we should only render these separators if the link actually has notes.